src/Core/Grid/Query/AttachmentQueryBuilder.php

```diff
@@ -41,26 +42,26 @@ final class AttachmentQueryBuilder extends AbstractDoctrineQueryBuilder
     private $searchCriteriaApplicator;
 
     /**
-     * @var string
+     * @var int
      */
-    private $employeeIdLang;
+    private $contextLanguageId;
 
     /**
      * @param Connection $connection
      * @param string $dbPrefix
      * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
-     * @param string $employeeIdLang
+     * @param LanguageContext $languageContext
      */
     public function __construct(
         Connection $connection,
         string $dbPrefix,
         DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
-        string $employeeIdLang
+        LanguageContext $languageContext
     ) {
         parent::__construct($connection, $dbPrefix);
 
         $this->searchCriteriaApplicator = $searchCriteriaApplicator;
-        $this->employeeIdLang = $employeeIdLang;
+        $this->contextLanguageId = $languageContext->getId();
```

Comment:
I don't think you need to store the value anymore, you can remove the field and replace it by storing the LanguageContext, and use `$this->languageContext()->getId()` instead

It's not just for the beauty of the code, it's also because the context services are lazy by default. I'm not 100% sure about how the lazyness is handled by Symfony, but I believe it is accessed JIT when it's used. So if you get the value in the constructor it forces the instanciation of the context service early when it can be delayed or maybe even not really be useful

Reply:
You can even use the property promotion in the constructor if you want:

```php
private LanguageContext $languageContext
```